gsutil.py

```diff
@@ -46,10 +46,11 @@ def _fix_google_module():
   """Reloads the google module to prefer our third_party copy.
 
   When Python is not invoked with the -S option, it can preload google module
-  via .pth file. After this happens, our third_party google
-  package may not in the __path__. After our third_party dependency directory is
-  put at the first place in the sys.path, google module should be reloaded
-  so that our vendored copy can be preferred.
+  via .pth file (https://docs.python.org/3/library/site.html). This means,
+  instead of using the google package from our "third_party" directory, it will end up
+  using the package from site_packages. To ensure
+  that the google package from our "third_party" directory is preferred,
+  we should insert the path at the start of sys.path and reload the google module.
```

Comment:
Still a bit tough to understand in its current form.
I'm pretty sure this shorter version works.
@dilipped can confirm.
```
When Python is *not* invoked with the -S option, it may preload the google module via .pth file.
This adds the "site_packages" version to "__path__" instead of the gsutil "third_party" version.
To force the "third_party" version, reload the google module.

This is a hacky. Reloading is required for the rare case that users have
google-auth already installed in their Python environment.
Note that this reload may cause an issue for Python 3.5.3 and lower
because of the weakref issue, fixed in Python 3.5.4:
https://github.com/python/cpython/commit/9cd7e17640a49635d1c1f8c2989578a8fc2c1de6.
```

I noticed Dilip's second version got rid of the term "vendored." This is good if it's intentional b/c we have a separate "vendored" folder with a specific meaning.

Reply:
>When Python is *not* invoked with the -S option, it may preload the google module via .pth file.

Based of your previous question , I think adding the link here would be helpful. I guess other people might be unaware of `.pth` as well and they would know where to read more about it.

>This adds the "site_packages" version to "__path__" instead of the gsutil "third_party" version.

I intentionally removed `__path__` out because it was not clear which `__path__` we are talking about. So I would probably change this to just say `"site_packages" version is preferred over gsutil "third_party" version`.

>To force the "third_party" version, reload the google module.`

This seems partially correct. You need to have the path present in front of `sys.path` before you reload and I think that's an important detail to mention here.

Reply:
Sounds good

Reply:
So:

> When Python is *not* invoked with the -S option, it may preload the google module via .pth file.
> This "site_packages" version is preferred over gsutil "third_party" version.
> To force the "third_party" version, insert the path at the start of sys.path and reload the google module.
> 
> This is a hacky. Reloading is required for the rare case that users have
> google-auth already installed in their Python environment.
> Note that this reload may cause an issue for Python 3.5.3 and lower
> because of the weakref issue, fixed in Python 3.5.4:
> https://github.com/python/cpython/commit/9cd7e17640a49635d1c1f8c2989578a8fc2c1de6.

Reply:
>This "site_packages" version is preferred over gsutil "third_party" version.

Maybe `This lead to the "site_packages" version being preferred over gsutil "third_party" version.